pkg/sqlutil/test_data_source.go

```diff
@@ -0,0 +1,85 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/proullon/ramsql/driver"
+	"github.com/stretchr/testify/require"
+)
+
+var _ DataSource = &TestDataSource{}
+
+// TestDataSource is a simple in-memory DataSource type which can be used for some types of unit testing.
+type TestDataSource struct {
+	db *sqlx.DB
```

Comment:
Is this wrapper type necessary? `*sqlx.DB` already implements `DataSource` directly.

Reply:
Not at present, although I was thinking we'll probably want to extend it to do other things in the future. For example, a test could add a hook that intercepts each request and gives a custom response... or modify the response in some way, or just assert some condition.

Reply:
I guess I could leave the constructor named the same but just return a `sqlx.DB` for now... that probably makes more sense, since we could still extend it later without modifying anything that's calling the constructor.

Reply:
I think we could leverage our existing sqlutil hook code later if we wanted to.